### ISSUE TITLE
HomePod Adapter

### ISSUE THREAD

**Issue by RolfM671** [ioBroker/AdapterRequests#577]
Hallo, würde mir einen Adapter zur Sprachausgabe auf dem HomePod wünschen, da meine Programmierkenntnisse dies nicht hergeben kann ich leider keinen selbst erstellen. Danke

**Comment by Apollon77** [comment]
Hast Du mal nach Libraries oder Möglichkeiten gesucht die es ggf gibt? ODer haben das andere Smart-Home Systeme?

**Comment by RolfM671** [comment]
ja, suche schon eine geraume Zeit in der Richtung gefunden habe ich aber noch nichts brauchbares für ioBroker, es gibt einige die auch danach suchen, gefunden habe ich nur etwas unter FHEM, hab mir schon überlegt ob ich auf FHEM umsteige.

**Comment by TheMBeat** [comment]
Hey @Apollon77,

[Maybe this could help.](https://github.com/mcfisto/node_airtunes2)

**Comment by RolfM671** [comment]
@TheMBeat 

yes, I have already installed airtunes, have already tested various things, but as I said, I simply lack the ability to program.

**Comment by RolfM671** [comment]
Hallo,
vielleicht hilft den Spezialisten hier dieser Artikel " https://forum.fhem.de/index.php?topic=102295.0 " weiter um etwas brauchbares für den ioBroker zu Entwickeln, wäre super.

**Comment by MrRo-de** [comment]
Hallo, ich habe ein Workaround den ich nutze um über Siri (HomePod) bestimmte Datenpunkte abfragen zu können.

https://github.com/MrRo-de/ioBroker-HomePod

Vielleicht hilft es ja jemanden weiter bis ein Adapter für den HomePod bzw. Siri kommt.

**Comment by RolfM671** [comment]
vielen Dank, da werde ich mich gleich mal damit auseinandersetzen. Im Augenblick versuche ich mich mit FHEM, da sollte ja Sprachausgabe über HomePod möglich sein leider ist das aber eine Oberfläche die Strukturell total zerrissen ist, läuft eine Anwendung und du integrierst eine neue dazu funktioniert die andere nicht mehr. Das ist eine Oberfläche für Vollzeitprogrammierer, da ist keine Funktionalität sondern Programmierkunst gefragt, also auch nicht das Wahre.

**Comment by MrRo-de** [comment]
@RolfM671 es wäre toll, wenn ich eine kurze Rückmeldung ob es geklappt hat und ob die Anleitung auch verständlich war bekommen würde.
Zu FHEM kann ich leider nichts sagen. Damit habe ich mich nie beschäftigt.

**Comment by RolfM671** [comment]
@MrRo-de sorry das ich mich noch nicht gemeldet habe, da ich mich noch um meine alte Kranke Mutter kümmern muss habe ich nur mal kurz in die Applikation geschaut aber noch nichts weiter getestet. Das weitere ist dass ich zwei Systeme teste um zu sehen welches meinen Anforderungen entspricht. Wenn ich mich wieder damit beschäftigen und testen kann werde ich auf jeden Fall Rückmeldung geben. Wünsche dir einen schönen Tag.

**Comment by x4N70pHyLL** [comment]
Hi, can I support somehow? Ist this still being  considered? I think for the people using Apple eco system, this is an essential. Probably it is not that much of work to include it in text2speech adapter or create an extra one.

My best guess is that it could be solved with an short cut (Kurzbefehle) on HomePod and a data point in ioBroker that gets filled with the text that should be spoken by Siri.
Challenges are probably
- how to trigger the short cut once there is text in the Data Point (the text itself could be pushed to the HomeKit world via a YAHKA adapter device/data point)
- how to empty the data point once it was spoken
- integrate in ioBroker (e.g. that Blockly has an option to "Send to Siri"

**Comment by DutchmanNL** [comment]
@x4N70pHyLL as far as I am aware there is not possibility in the HomeKit eco system to send a text to speak to a device.
The workaround here mentioned is also what I use, just create an alias/dummy ad implement it by Homekit (Yahka)

**Comment by x4N70pHyLL** [comment]
@DutchmanNL Yes it seems not directly but doesn't it make sense to "professionalize" the workaround a little bit to enable others to integrate it in their adapters like Blockly etc.
At least for me it is not very clear how to set up a Yahka device (which type etc) and how to pass a string from a data point to the device. Then there needs to be a trigger inside HomeKit/Shortcut/Kurzbefehle like "if (switch in device is pressed) {let Siri speak the atribute x of the device}"

**Comment by DutchmanNL** [comment]
Of its only related to the yahka objects I have a automation solution for that which my fit. In generic it's very hard to make a separate "will always word" code for this specific case but happy to brainstorm if my automation fits the needs. But would prefer to do this on discord or telegram if that's ok for you ?

**Comment by RolfM671** [comment]
@MrRo-de So, nach langer langer Zeit habe ich mich nun endlich mit deinem Workaround befassen können, ich muß sagen dass ich einige Mühe hatte da ich mit Programmierung rein gar nichts anfangen kann. Dennoch hat es geklappt und ich habe einen Test mit Tankerkönig zum funktionieren gebracht, also wollte ich das selbe mit dem Oilfox zur Ausgabe des Füllstands vom Heizöltank machen und habe die vorhandenen Daten dafür abgeändert. Leider funktioniert es mit dem Oilfoxadapter nicht, keine Ahnung warum, es gibt keine Fehlermeldungen.
Auf jeden Fall möchte ich mich für dein Workaround herzlich bedanken, einen richtigen Adapter wie bei Alexa kann dieser Workaround dennoch nicht ersetzen,

**Comment by DutchmanNL** [comment]
> Auf jeden Fall möchte ich mich für dein Workaround herzlich bedanken, einen richtigen Adapter wie bei Alexa kann dieser Workaround dennoch nicht ersetzen,

hmm, das macht im gründe Yahka... aber der wizard bzw automatische Zuweisung fehlt dir wie beim iOT ?

**Comment by x4N70pHyLL** [comment]
@DutchmanNL sure we can do it privately, but I think the general solution (it seems to be the best we have at the moment) should be interesting for everyone, right?

Like the workaround postet above, but how to set up including some transfer mechanism to shift Strings over to some Yahka device and the automation from Apple to make Siri read out the String.

**Comment by DutchmanNL** [comment]
> @DutchmanNL sure we can do it privately, but I think the general solution (it seems to be the best we have at the moment) should be interesting for everyone, right?
> 
> 
> 
> Like the workaround postet above, but how to set up including some transfer mechanism to shift Strings over to some Yahka device and the automation from Apple to make Siri read out the String.

The outcome will be for everyone now worries, but it's a very prill alpha currently 😅😋

**Comment by x4N70pHyLL** [comment]
@DutchmanNL Ok got it, just let me know how/where, Telegram is fine for me, but I did not do anything with it regarding ioBroker/Github. You can find me under the same username

**Comment by haselchen** [comment]
@DutchmanNL 
@Apollon77 

Wird an der Umsetzung noch gearbeitet?

**Comment by RolfM671** [comment]
**@haselchen**
Ich denke nicht dass an der Umsetzung gearbeitet wird, dazu fehlt die Nachfrage von Anwendern.

**Comment by haselchen** [comment]
> **@haselchen** Ich denke nicht dass an der Umsetzung gearbeitet wird, dazu fehlt die Nachfrage von Anwendern.

Keine Ahnung wieviele Daumen hoch Du brauchst....ich habe jedenfalls einen hinterlassen

**Comment by Apollon77** [comment]
@haselchen Status ist "Requested" ... es halt also noch keiner damit gestartet ... Kommt vllt noch und ja die Hardware muss man dafür auch haben

**Comment by reiterth** [comment]
Der workaround funktioniert scheinbar mit der aktuellen Homekit Version nicht mehr

**Comment by Draexl** [comment]
Hi, gibt es hier Updates? Hat irgendwer die SimpleAPI noch am laufen? Überlege mir die Homepods zuzulegen, aber nur wenn es eine funktionierende Lösung gibt.
